ansible/readme.md

```diff
@@ -0,0 +1,56 @@
+# self-hosted kubernetes cluster with bootkube and ansible
+This is an ansible installer which brings up full kubernetes cluster with the help of bootkube. 
+## how to use
+
+Edit `hosts` and fill in the desired instances. Run the following to bring up all at once:
+
+```
+ansible-playbook -i hosts site.yml
+```
+
+If you want to target specific group of instances, read on
+
+### bootstrap master
+
+```
+ansible-playbook -i hosts site.yml -l bootstrap
+```
+
+### join instances to control-plane
+
+```
+ansible-playbook -i hosts site.yml -l master
+```
+
+### join worker nodes
+```
+ansible-playbook -i hosts site.yml -l workers
+```
+
+### connect with kubectl
+the bootstrap playbook will copy a kubeconfig to your ansible folder: `roles/kubelet/templates/kubeconfig`. You can use it to connect to the cluster:
+```
+kubectl --kubeconfig=roles/kubelet/templates/kubeconfig cluster-info
+```
+
+### scale etcd
+kubectl apply doesn't work for TPR at the moment. See https://github.com/kubernetes/kubernetes/issues/29542. As a workaround, we use cURL to resize the cluster.
+
+```
+kubectl --kubeconfig=roles/kubelet/templates/kubeconfig --namespace=kube-system get cluster.etcd kube-etcd -o json > etcd.json && \
+vim etcd.json && \
```

Comment:
```
    "spec": {
        "selfHosted": {
            "bootMemberClientEndpoint": "http://10.7.183.59:12379"
        },
        "size": 3,
        "version": "3.1.0"
    },
```
the `size` parameter needs to be set to the desired size. Use 1, 3, 5, etc.

Until the TPR issue is fixed (I think with 1.6 or maybe even 1.5.4) this needs to be done this way. It can happen that Kubernetes will not accept the manual push with cURL (resulting in 409). The process needs to be repeated then.